test/test.js

```diff
@@ -1,8 +1,195 @@
+
+	// Check the observation’s dependency data
+	var observationDepData = canReflectDeps.getDependencyDataOf(observation);
+	assert.notOk(observationDepData.whatIChange, "the observation does not change anything");
+	assert.notOk(observationDepData.whatChangesMe.mutate, "the observation does not have mutation dependencies");
+	assert.notOk(
+		observationDepData.whatChangesMe.derive.valueDependencies,
+		"the observation has zero derive.valueDependencies"
+	);
+	assert.equal(
+		observationDepData.whatChangesMe.derive.keyDependencies.size,
+		2,
+		"the observation has two derive.keyDependencies"
+	);
+	assert.ok(
+		observationDepData.whatChangesMe.derive.keyDependencies.get(outer).has("inner"),
+		"the observation changes outer’s 'inner' property"
+	);
+	assert.ok(
+		observationDepData.whatChangesMe.derive.keyDependencies.get(outer.get("inner")).has("key"),
+		"the observation changes outer.inner’s 'key' property"
+	);
+
+	// Check outer.inner’s dependency data
+	var innerDepData = canReflectDeps.getDependencyDataOf(outer, "inner");
+	assert.notOk(innerDepData.whatChangesMe, "outer.inner is not changed by anything");
+	assert.notOk(innerDepData.whatIChange.mutate, "outer.inner does not have mutate dependencies");
+	assert.notOk(
+		innerDepData.whatIChange.derive.keyDependencies,
+		"outer.inner has zero derive.keyDependencies"
+	);
+	assert.equal(
+		innerDepData.whatIChange.derive.valueDependencies.size,
+		1,
+		"outer.inner has one derive.valueDependencies"
+	);
+	assert.ok(
+		innerDepData.whatIChange.derive.valueDependencies.has(observation),
+		"outer.inner changes the observation"
+	);
+
+	// Check outer.inner.key’s dependency data
+	var keyDepData = canReflectDeps.getDependencyDataOf(outer.get("inner"), "key");
+	assert.notOk(keyDepData.whatChangesMe, "outer.inner.key is not changed by anything");
+	assert.notOk(keyDepData.whatIChange.mutate, "outer.inner.key does not have mutate dependencies");
+	assert.notOk(
+		keyDepData.whatIChange.derive.keyDependencies,
+		"outer.inner.key has zero derive.keyDependencies"
+	);
+	assert.equal(
+		keyDepData.whatIChange.derive.valueDependencies.size,
+		1,
+		"outer.inner.key has one derive.valueDependencies"
+	);
+	assert.ok(
+		keyDepData.whatIChange.derive.valueDependencies.has(observation),
+		"outer.inner.key changes the observation"
+	);
+});
+
+QUnit.test("to method works", function() {
+	var outer = {inner: {key: "hello"}};
+	var setProp = canValue.to(outer, "inner.key");
+
+	// Getting the value shouldn’t work
+	QUnit.equal(canReflect.getValue(setProp), undefined, "getting doesn’t work");
+
+	// Test setting the value
+	canReflect.setValue(setProp, "aloha");
+	QUnit.equal(outer.inner.key, "aloha", "setting works");
+});
+
+QUnit.test("to method returns an observable with a helpful name", function() {
+	var outer = {inner: {key: "hello"}};
+	var observable = canValue.to(outer, "inner.key");
+	QUnit.equal(
+		canReflect.getName(observable),
+		"CanValueToObservable<Object{}.inner.key>",
```

Comment:
Thoughts on the names for these?

Note that I didn’t do anything special for KeyObservable’s name… right now it is just `KeyObservable<>` but I will fix that in can-key to be something like `KeyObservable<Object{}.inner.key>`